lib/Models/GeoJsonCatalogItem.ts

```diff
@@ -0,0 +1,195 @@
+import { computed, observable } from "mobx";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import DataSource from "terriajs-cesium/Source/DataSources/DataSource";
+import defined from "terriajs-cesium/Source/Core/defined";
+import GeoJsonCatalogItemTraits from "../Traits/GeoJsonCatalogItemTraits";
+import GeoJsonDataSource from "terriajs-cesium/Source/DataSources/GeoJsonDataSource";
+import GetCapabilitiesMixin from "../ModelMixins/GetCapabilitiesMixin";
+import Mappable from "./Mappable";
+import Model from "./Model";
+import TerriaError from "../Core/TerriaError";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import when from "terriajs-cesium/Source/ThirdParty/when";
+
+var loadBlob = require("../Core/loadBlob");
+var loadJson = require("../Core/loadJson");
+var proxyCatalogItemUrl = require("./proxyCatalogItemUrl");
+var readJson = require("../Core/readJson");
+var { updateModelFromData, getJson } = require("./loadGeoJson");
+var zip = require("terriajs-cesium/Source/ThirdParty/zip");
+
+var zipFileRegex = /.zip\b/i;
+var geoJsonRegex = /.geojson\b/i;
+
+export default class GeoJsonCatalogItem
+    extends UrlMixin(CatalogMemberMixin(Model(GeoJsonCatalogItemTraits)))
+    implements Mappable {
+    @observable dataSource?: GeoJsonDataSource;
+
+    @observable
+    show: boolean = true;
+
+    data: any;
+
+    @computed
+    get mapItems() {
+        if (this.dataSource === undefined) {
+            return [];
+        }
+        this.dataSource.show = this.show;
+        return [this.dataSource];
+    }
+
+    loadMetadata(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    loadData(): Promise<void> {
+        let that = this;
+        this.dataSource = new GeoJsonDataSource(this.name);
```

Comment:
We (eventually) need to guard against repeated invocations of `loadData` (i.e. it needs to be harmless to call it extra times). I just looked at WMSCatalogItem and it seems to have the same problem, though, so don't worry about it for now.